apps/hash-frontend/src/pages/drafts.page/draft-entity/draft-entity-viewers.tsx

```diff
@@ -0,0 +1,114 @@
+import { FunctionComponent, useMemo } from "react";
+
+import { useOrgsWithLinks } from "../../../components/hooks/use-orgs-with-links";
+import { useUsers } from "../../../components/hooks/use-users";
+import {
+  GetEntityAuthorizationRelationshipsQuery,
+  GetEntityAuthorizationRelationshipsQueryVariables,
+} from "../../../graphql/api-types.gen";
+import { getEntityAuthorizationRelationshipsQuery } from "../../../graphql/queries/knowledge/entity.queries";
+import { MinimalUser } from "../../../lib/user-and-org";
+
+export const DraftEntityViewers: FunctionComponent<{
+  entity: Entity;
+}> = ({ entity }) => {
+  const { data } = useQuery<
+    GetEntityAuthorizationRelationshipsQuery,
+    GetEntityAuthorizationRelationshipsQueryVariables
+  >(getEntityAuthorizationRelationshipsQuery, {
+    variables: { entityId: entity.metadata.recordId.entityId },
+    fetchPolicy: "cache-and-network",
+  });
+
+  const authorizationRelationships = data?.getEntityAuthorizationRelationships;
+
+  const { users } = useUsers();
+
+  const { orgs: orgViewers } = useOrgsWithLinks({
+    orgAccountGroupIds: authorizationRelationships
+      ?.map(({ subject }) =>
+        subject.__typename === "AccountGroupAuthorizationSubject"
+          ? subject.accountGroupId
+          : [],
+      )
+      .flat(),
+  });
+
+  const orgMemberViewers = useMemo(() => {
+    if (orgViewers) {
+      return orgViewers
+        .map(({ memberships }) => memberships.map(({ user }) => user))
+        .flat();
+    }
+  }, [orgViewers]);
+
+  const userViewers = useMemo(() => {
+    if (authorizationRelationships && users) {
+      return authorizationRelationships.reduce<MinimalUser[]>(
+        (prev, { subject }) => {
+          if (subject.__typename === "AccountAuthorizationSubject") {
+            const user = users.find(
+              ({ accountId }) => accountId === subject.accountId,
+            );
+
+            if (
+              user &&
+              !prev.some(({ accountId }) => accountId === user.accountId)
+            ) {
+              return [...prev, user];
+            }
+          }
+
+          return prev;
+        },
+        [],
+      );
+    }
+  }, [authorizationRelationships, users]);
+
+  const copy = useMemo(() => {
+    if (authorizationRelationships) {
+      if (
+        authorizationRelationships.some(
+          ({ subject }) => subject.__typename === "PublicAuthorizationSubject",
+        )
+      ) {
+        return "Visible to anyone";
+      }
+
+      if (authorizationRelationships.length === 1) {
+        return "Only visible to me";
+      }
+
+      if (userViewers && orgMemberViewers) {
+        const viewers = [...userViewers, ...orgMemberViewers].filter(
+          (viewer, index, all) =>
+            all.findIndex(({ accountId }) => accountId === viewer.accountId) ===
+            index,
+        );
+
+        return (
+          <>
+            Visible to <strong>{viewers.length} people</strong>
+          </>
+        );
```

Comment:
This is a design question for @nonparibus for follow-up, but I wonder if it's better to say 'viewable to members of X' somehow here, rather than just counting the members and leaving it to the user to figure out that this represents the membership count of the org

Reply:
I'm not sure the number here will always refer to all members of the org. It may be a subset who can view entities of a certain type, etc. Ideally on hover over the number we would show which users have access in a tooltip.

Reply:
I've created [this follow-up ticket](https://linear.app/hash/issue/H-1550/display-tooltip-over-visible-to-x-people-to-indicate-whether-those-are) to track this, may require some designs if there are lots of users, or we want to include information about whether this has been shared with an org or specific users.